setup.py

```diff
@@ -59,6 +59,7 @@
     'azure-storage-blob>=12.0.0,<13',
     'azure-storage-file-datalake>=12.11.0,<13',
     'azure-identity>=1.13.0',
+    'pyspark==3.4.1',
```

Comment:
Does this need to be a hard import? We have so many dependencies already...

Reply:
@XiaohanZhangCMU Why import only if the relevant functions are called? Like do we to install a hard coded version of PySpark for all of our users?

Reply:
I am also worried about this downgrading a users databricks' version of PySpark and breaking everything later. At the very least we shouldn't be tying it to a specific minor version, espically given how stable PySpark is.

Reply:
I moved pyspark to extra_deps['databricks']. I also cut loose the spark versions to next release, it should be fine.